### ISSUE TITLE
Version not being passed to zircote/swagger-php

### ISSUE THREAD

**Issue by bramslob** [nelmio/NelmioApiDocBundle#1950]
*TLDR* please pass the defined version (if applicable) through to swagger-php

With the new update of Swagger-php to 4.2.* (https://github.com/zircote/swagger-php/releases/tag/4.2.0) it now defaults to OpenApi 3.0.0, with the option to go to OpenApi 3.1.0 when setting the openapi version explicitly.

This mostly works. When setting the following yaml config in `nelmio_api_doc.yaml`:
```yaml
nelmio_api_doc:
    documentation:
        openapi: 3.1.0
        # the rest of the config
```

the top of the doc.json will show openapi 3.1.0

```json
{
    "openapi": "3.1.0"
    // the rest of the json
}
```

However, one of the features/fixes that was added is to have support for having `description` next to `$ref`.
This is still there in the OpenApi\AbstractNotation.php class (vendor/zircote/swagger-php/src/Annotations/AbstractAnnotation.php:346), however, it is put behind a version flag:

```php
        // $ref
        if (isset($data->ref)) {
            // Only specific https://github.com/OAI/OpenAPI-Specification/blob/3.1.0/versions/3.1.0.md#reference-object
            $ref = ['$ref' => $data->ref];
            if ($this->_context->version == OpenApi::VERSION_3_1_0) {
                $defaultValues = get_class_vars(get_class($this));
                foreach (['summary', 'description'] as $prop) {
                    if (property_exists($this, $prop)) {
                        if ($this->$prop !== $defaultValues[$prop]) {
                            $ref[$prop] = $data->$prop;
                        }
                    }
                }
            }
            $data = (object) $ref;
        }
```

The problem is with the `$this->_context->version` part. The new version of the swagger-php package now has this concept of a version, but the nelmio code never passes the defined version into it. Therefore, the above check fails.
This was verified locally by removing the `if ($this->_context->version == OpenApi::VERSION_3_1_0)` conditional and then the `description` showed up next to `$ref`.

The logical solution for this seems to pass the openapi config into the swagger-php code. Maybe this is already possible with the current code, but I have not seen an option somewhere to do so.

**Comment by DerManoMann** [comment]
`swagger-php` v4.2.14 introduced some different behaviour for this in that the version set on `@OA\OpenApi` is being used **unless* a version is also specified via the `Generator`.
In other words: unless a version is forced via `Generator::setVersion` the version of `@OA\OpenApi` is used (this still defaults to 3.0.0)

**Comment by DerManoMann** [comment]
You should be able to set the version on your global `@OA\OpenApi` annotation. If you do not have one this can be next to your `@OA\Info` or even have the info nested:
```php
/**
 * @OA\OpenApi(
 *     openapi="3.1.0",
 *     @OA\Info(...)
 * )
 */
```

**Comment by GuilhemN** [comment]
Closing as the solution provided by @DerManoMann should fix the original issue (requires `swagger-php > 4.2.14`).
